File: 202612/1210/Main.java
import java.io.*;
import java.util.Arrays;

public class Main {
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int tc = Integer.parseInt(br.readLine());
        while(tc >0){
            int n = Integer.parseInt(br.readLine());
            int[] arr = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
            int maxStock = 0;
            long profit = 0L;
            for(int i= arr.length-1; i>=0; --i){
                if(maxStock < arr[i]){
                    maxStock = arr[i];
                } else {
                    profit = profit + (maxStock - arr[i]);
                }
            }

            System.out.println(profit);
            tc --;
        }


    }
}

File: 202612/1210/main.py
# https://www.acmicpc.net/problem/11501

tc = int(input())
while tc:
    n = int(input())
    stocks = list(map(int, input().split(" ")))
    maxStock = 0
    profit = 0

    for i in range(len(stocks) - 1, -1, -1):
        if maxStock < stocks[i]:
            maxStock = stocks[i]
        else:
            profit = profit + (maxStock - stocks[i])

    print(profit)

    tc -= 1


